GetSubscriptionKeys/handler.ts

```diff
@@ -0,0 +1,178 @@
+import { Context } from "@azure/functions";
+
+import * as express from "express";
+
+import { getRequiredStringEnv } from "io-functions-commons/dist/src/utils/env";
+import {
+  AzureApiAuthMiddleware,
+  IAzureApiAuthorization,
+  UserGroup
+} from "io-functions-commons/dist/src/utils/middlewares/azure_api_auth";
+import { ContextMiddleware } from "io-functions-commons/dist/src/utils/middlewares/context_middleware";
+import {
+  withRequestMiddlewares,
+  wrapRequestHandler
+} from "io-functions-commons/dist/src/utils/request_middleware";
+
+import { ApiManagementClient } from "@azure/arm-apimanagement";
+import { RestError } from "@azure/ms-rest-js";
+import * as msRestNodeAuth from "@azure/ms-rest-nodeauth";
+import { ApplicationTokenCredentials } from "@azure/ms-rest-nodeauth";
+import { TokenResponse } from "@azure/ms-rest-nodeauth/dist/lib/credentials/tokenClientCredentials";
+import { right, toError } from "fp-ts/lib/Either";
+import { fromEither, TaskEither, tryCatch } from "fp-ts/lib/TaskEither";
+import {
+  IResponseErrorInternal,
+  IResponseErrorNotFound,
+  IResponseSuccessJson,
+  ResponseErrorInternal,
+  ResponseErrorNotFound,
+  ResponseSuccessJson
+} from "italia-ts-commons/lib/responses";
+import { ServiceId } from "../generated/definitions/ServiceId";
+import { ServiceIdMiddleware } from "../utils/middlewares/serviceid";
+
+const azureSubscriptionId = getRequiredStringEnv("AZURE_SUBSCRIPTION_ID");
+const servicePrincipalClientId = getRequiredStringEnv(
+  "SERVICE_PRINCIPAL_CLIENT_ID"
+);
+const servicePrincipalSecret = getRequiredStringEnv("SERVICE_PRINCIPAL_SECRET");
+const servicePrincipalTenantId = getRequiredStringEnv(
+  "SERVICE_PRINCIPAL_TENANT_ID"
+);
+const azureApimResourceGroup = getRequiredStringEnv(
+  "AZURE_APIM_RESOURCE_GROUP"
+);
+const azureApim = getRequiredStringEnv("AZURE_APIM");
+
+export interface ITokenAndCredentials {
+  readonly token: TokenResponse;
+  readonly loginCreds: ApplicationTokenCredentials;
+  readonly expiresOn: number;
+}
+
+export interface IServicePrincipalCreds {
+  readonly servicePrincipalClientId: string;
+  readonly servicePrincipalSecret: string;
+  readonly servicePrincipalTenantId: string;
+}
+
+function loginToApim(
+  servicePrincipalCreds: IServicePrincipalCreds,
+  tokenAndCredentials?: ITokenAndCredentials
+): TaskEither<Error, ITokenAndCredentials> {
+  const isTokenExpired = tokenAndCredentials
+    ? tokenAndCredentials.expiresOn <= Date.now()
```

Comment:
please remove all the code related to token caching since this is already handled inside ms-rest-azure:

https://github.com/Azure/azure-sdk-for-node/blob/8691046c2ca738552d815f6967f5f54b04840276/runtime/ms-rest-azure/lib/credentials/applicationTokenCredentials.js#L129

just call `loginWithServicePrincipal`